State.py

```diff
@@ -1,119 +1,108 @@
-import server
 import threading
 import RaftMessages_pb2 as protoc
 
 class State():
-  def __init__(self, termNumber, server):
-    self.termNumber = termNumber  
-    self.server = server
+  def __init__(self):
+    self.termNumber = 0 
+
+  def replyAENACK(self, toAddr, toPort, termNumber):
+    message = protoc.AppendReply()
+    message.toAddr = toAddr
+    message.toPort = toPort
+    message.term = termNumber
+    message.success = False
+    return protoc.APPENDREPLY, message
+
+  def replyAEACK(self, toAddr, toPort, termNumber):
+    message = protoc.AppendReply()
+    message.toAddr = toAddr
+    message.toPort = toPort
+    message.term = termNumber
+    message.success = True
+    return protoc.APPENDREPLY, message
 
   def sendVoteNACK(self, toAddr, toPort, termNumber):
-    print("Send VoteReply NACK to ", end="")
     voteack = protoc.VoteResult()
     voteack.toAddr = toAddr
     voteack.toPort = toPort
     voteack.term = termNumber
     voteack.granted = False
-    self.server.talk(protoc.VOTERESULT, voteack)
+    return protoc.VOTERESULT, voteack
 
   def sendVoteACK(self, toAddr, toPort, termNumber):
-    print("Send VoteResult ACK to ", end="")
     voteack = protoc.VoteResult()
     voteack.toAddr = toAddr
     voteack.toPort = toPort
     voteack.term = termNumber
     voteack.granted = True
-    self.server.talk(protoc.VOTERESULT, voteack)
+    return protoc.VOTERESULT, voteack
 
-  def replyAENACK(self, toAddr, toPort, termNumber):
-    print("Send AppendReply NACK to ", end="")
-    message = protoc.AppendReply()
-    message.toAddr = toAddr
-    message.toPort = toPort
-    message.term = termNumber
-    message.success = False
-    self.server.talk(protoc.APPENDREPLY, message)
-
-  def replyAEACK(self, toAddr, toPort, termNumber):
-    print("Send AppendReply ACK to ", end="")
-    message = protoc.AppendReply()
-    message.toAddr = toAddr
-    message.toPort = toPort
-    message.term = termNumber
-    message.success = True
-    self.server.talk(protoc.APPENDREPLY, message)   
 
 class LeaderState(State):
-  def __init__(self, termNumber, server):
-    State.__init__(self, termNumber, server)
+  def __init__(self):
+    State.__init__(self)
+    # Send Heartbeats once transitioned to inform a leader is present
     self.sendHeartbeat()
```

Comment:
I think this may be a bug.  The sendHeartbeat() method doesn't actually send anything. It returns the message type and the message.  This call is ignoring the return value so unless I'm missing something this won't actually send a heartbeat.